src/main/java/com/iota/iri/hash/Kerl.java

```diff
@@ -6,142 +6,122 @@
 import java.math.BigInteger;
 import java.security.DigestException;
 import java.util.Arrays;
+import java.util.stream.IntStream;
+
+public final class Kerl implements Sponge {
 
-public class Kerl implements Sponge {
     public static final int BIT_HASH_LENGTH = 384;
     public static final int BYTE_HASH_LENGTH = BIT_HASH_LENGTH / 8;
 
-    private byte[] byte_state;
-    private int[] trit_state;
```

Comment:
The instance state has been removed and replaced with method-local state. It make reading the processing much easier and there is no performance hit.